design/alameda/design.md

```diff
@@ -0,0 +1,34 @@
+## What is Alameda
+
+Alameda is an intelligent resource orchestrator for Kubernetes, providing the features of autonomous balancing, scaling, and scheduling by using machine learning. Alameda learns the continuing changes of computing resources from K8S clusters, predicts the future computing resources demands for pods and nodes, and intelligently orchestrates the underlying computing resources without manual configuration.
+
+For more details, please refer to https://github.com/containers-ai/Alameda
+
+## How Alameda works with Rook
+
+![integration](./architecture_rook_integration.png)
+(Note: CRD AlamedaResource is called AlamedaScaler and CRD AlamedaResourcePrediction is called AlamedaRecommendation from release 0.2)
+
+This figure illustrates Alameda's architecture and how it can work with Rook. The primary purpose of Alameda is to **recommend optimal computing resource configuration for Kubernetes**. To achieve that, users of Alameda (which is Rook here) specify which Pods require metrics prediction and resource configuration recommendation by creating *AlamedaResource* CRs ([example](https://github.com/containers-ai/alameda/blob/master/example/samples/nginx/alameda_deployment.yaml)). After some machine learning calculations, users can see metrics predictions and recommendations in the *AlamedaResourcePrediction* CRs. An example can be found [here](https://github.com/containers-ai/alameda/blob/master/docs/quickstart.md#example). Please note that starting from Alameda release 0.2, the metrics predictions will be store in a time-series DB instead of CRs due to performance and size considerations of etcd.
```

Comment:
> the metrics predictions will be store in a time-series DB

What does the interface look like for an orchestrator to consume the predictions and recommendations? does it call a specific Alameda API?  how does it discover that endpoint?

How big are the predictions and recommendations?  I can see Alameda gathering a lot of metrics/stats and that taking up a lot of space, but I would expect the recommendations to have a small footprint comparatively. Maybe a CRD approach could still work if the predictions are small and each CRD is over time updated with the latest recommendations, as opposed to new recommendation CRDs constantly being generated.